controllers/gcpcluster.go

```diff
@@ -71,9 +74,10 @@ func (r *GCPClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		log.Info("GCP Cluster does not have an owner cluster yet")
 		return ctrl.Result{}, nil
 	}
+
 	if gcpCluster.Status.Network.SelfLink == nil || *gcpCluster.Status.Network.SelfLink == "" {
 		log.Info("GCP Cluster does not have network set yet")
-		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 30}, nil
+		return ctrl.Result{}, nil
```

Comment:
There is no need to requeue here. Once the status is updated with the `selfLink`, a reconciliation look is triggered.